src/ComputedFieldTypes.jl

```diff
@@ -161,8 +161,13 @@ function rewrite_new!(e::Expr, tname::Symbol, decl_tvars, def)
                 pfname[1] = make_Type_expr(tname, decl_tvars)
                 curly = e.args[1].args[1]
                 if !(isa(curly, Expr) && curly.head === :curly)
-                    curly = Expr(:curly, curly)
-                    e.args[1].args[1] = curly
+                    curly = Expr(:where, e.args[1])
```

Comment:
this seems wrong: I don't think you want to write `S{T}() where {T}` into `(S where T)() where {T}`

Reply:
The main point is that the nested expression of 0.6 with form `Expr(:curly,Expr(:call,...),...)` needs to be replaced with the reverse ordering for 1.0 with `Expr(:where,Expr(:call,...),...)`.

The changes I made effectively make the `:call` expression trade places with `:curly` and, instead of having `:curly` you now have `:where`, but traded in places with `:call`.

You will notice, I am using `e.args[1]` instead of `e.args[1].args[1]` to trade the places.

Reply:
Are you doing this as a deprecation upgrade?